### DISCUSSION TITLE
Site Management: Import/Export/Templates for Sites

### DISCUSSION THREAD

**Discussion by thabaum** [oqtane/oqtane.framework#3442] [Ideas]
Idea of Import/Export of sites and creating a template from a site.

I would like to use a site for a template, export sites to other sites to be imported and saving sites as a template that can be used when you add a site.  Site templates would need a way to be manages as well for when they are bugged out, irrelevant or no longer work you can delete it.

Also deleting sites is something else that would be of value.   A host being able to configure the hosting model of a site in the event it is using the wrong one and no longer accessible such as Hybrid.  It would be good idea I think for some important settings that can lock a host user out be in the Site Management page where you can inline edit maybe with the pencil icon like used for page editing but on the lines that a site is on making the Url and Hosting Model editable.

Issues relating should be made for any future enhancements worthy of adding to the framework to keep on the logs as something that lets others know it is being reviewed for possible inclusion down the road.  Even if they are not handled in the next 3-6 months an issue can still marinate for when time comes to be discussed.

**Comment by sbwalker** [comment]
This is a good example of an area where Oqtane does not intend to emulate DNN. The serialization of sites/pages/modules in DNN was hugely problematic to develop/test/maintain over time. As a result Oqtane is taking a different approach in this area. Since Oqtane's primary audience is software developers and not content managers, it is providing extensibility points, APIs, and interfaces to allow developers to automate the creation of sites/pages/modules ie. ISiteTemplate, ISiteMigration, and routable modules.
